src/bullsandcows/Main.java

```diff
@@ -1,7 +1,194 @@
-package bullsandcows;
+
+import java.util.*;
+//package bullsandcows;
 
 public class Main {
+
+    public static Set<Character> secretGenerator(int numbers, int possibleNumbers) {
+        /*
+        We will generate ints and then convert them to valid characters from ascii table.
+
+        ASCII codes:
+
+        0-9: 48-57 (max possibleNumbers - 10)
+        a-z: 97-122 (max possibleNumbers - 36)
+        A-Z: 65-90 (max possibleNumbers - 62) - it makes game too complicated, but can be added at any moment
+
+         */
+        int initialNumbers = numbers;
+
+        Set<Character> answer = new LinkedHashSet<>();
+
+        if (numbers > possibleNumbers | numbers < 1 | possibleNumbers < 1)
+        {
+            System.out.printf("Can't generate a secret number with a length of %d with only %d unique symbols.%n", numbers, possibleNumbers);
+            System.out.println("Please enter a number of possible symbols no less than a number of symbols in secret.");
+            return answer;
+
+        } else
+        {
+
+            Random random = new Random(1);
+
+            while (numbers > 0)
+            {
+                char digit = encodeInt(random.nextInt(possibleNumbers));
+
+                // If answer already contains that digit - pass.
```

Comment:
This comment is not very useful, because the expression below is pretty clear and says the same thing. It is better to comment snippets which are not very clear or very specific. 

PS: good code does not (almost) need extra comments. It is self-documented and can be read as a book :)